### ISSUE TITLE
alt is considered NodeValue::Text

### ISSUE THREAD

**Issue by dev0Guy** [kivikakk/comrak#281]
HI . 
converted the following md file 
`# Awesome README [![Awesome](https://cdn.jsdelivr.net/gh/sindresorhus/awesome@d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)](https://github.com/sindresorhus/awesome#readme)
`
Created node iter on type nodevalue::Text. while debuging i saw the alt="text" is considered text. is there a way to discard it ?? 

Thanks in advance,
dev0guy

**Comment by dev0Guy** [comment]
`use std::{fs};
use comrak::{parse_document,format_html, Arena, ComrakOptions, nodes::{AstNode, NodeValue}};

fn iter_nodes<'a,F>(node: &'a AstNode<'a>, f: &F)
    where F: Fn(&'a AstNode<'a>){
    f(node);
    for c in node.children(){
        iter_nodes(c, f);
    }

}

fn main(){
    let markdown_input = fs::read_to_string("./assets/readme.md").expect("Can't open file");
    // let markdown_input = "This is my input.\n\n1. Also my input.\n2. Certainly my input. \n ";
    let arena = Arena::new();
    let root = parse_document(&arena, &markdown_input, &ComrakOptions::default());

    let mut html = vec![];
    // println!("{}",String::from_utf8(html).unwrap());

    iter_nodes(root, &|node|{
        // println!("{:?}", node)
        match &mut node.data.borrow_mut().value{
            &mut NodeValue::Text(ref mut text) =>{
                let orig = std::mem::replace(text, vec![]);
                let inner_text = String::from_utf8(orig).unwrap();
                let new_line = format!(r#" {{"{}"}} "#, &inner_text);
                *text = new_line.as_bytes().to_vec();
            }
            _ => (),
        }
    });
    format_html(root, &ComrakOptions::default(), &mut html).unwrap();
    let as_string = String::from_utf8(html).unwrap();//.replace("&quot;", r#"""#);
    println!("{}",as_string);

}`


output: 
`<h1> {&quot;Awesome README &quot;} <a href="https://github.com/sindresorhus/awesome#readme"><img src="https://cdn.jsdelivr.net/gh/sindresorhus/awesome@d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg" alt=" {&quot;Awesome&quot;} " /></a></h1>`

**Comment by kivikakk** [comment]
Indeed, alt text is considered a text node nested inside an image in CommonMark; see the AST output from the [interactive dingus](https://spec.commonmark.org/dingus/?text=%23%20Awesome%20README%20%5B!%5BAwesome%5D(https%3A%2F%2Fcdn.jsdelivr.net%2Fgh%2Fsindresorhus%2Fawesome%40d7305f38d29fed78fa85652e3a63e154dd8e8829%2Fmedia%2Fbadge.svg)%5D(https%3A%2F%2Fgithub.com%2Fsindresorhus%2Fawesome%23readme)%20):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">
<document xmlns="http://commonmark.org/xml/1.0">
  <heading level="1">
    <text>Awesome README </text>
    <link destination="https://github.com/sindresorhus/awesome#readme" title="">
      <image destination="https://cdn.jsdelivr.net/gh/sindresorhus/awesome@d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg" title="">
        <text>Awesome</text>
      </image>
    </link>
  </heading>
</document>
```

See also the [Images](https://spec.commonmark.org/0.30/#images) part of the spec.

If you don't want to consider it in your iterator, you'll need to note when you're nested inside an image and not process if so.

**Comment by dev0Guy** [comment]
@kivikakk  Thanks for the answer!
How can I note when you're nested inside an image ?


Thanks in advance!

**Comment by kivikakk** [comment]
Seeing as this isn't an issue with Comrak itself, I'm going to convert this to a discussion. I don't have the time right now to write this up, as it's more of a general programming question, but I may later (or somebody else might be able to step in and help).
